src/helpers/report-helpers.ts

```diff
@@ -56,8 +57,18 @@ export class ReportHelpers {
     })
 
     // fetch the raw content
-    const contentResponse = await axios.get(documentResponse.data.url)
-    const rawData = contentResponse.data
+    const contentResponse = await axios.get(documentResponse.data.url, {
+      responseType: 'arraybuffer',
+    })
+
+    let rawBuffer = contentResponse.data
+    // amazon sometimes encodes the response as gzip and specifies the compressionAlgorithm
+    // we want to decode the data and make this process transparent to the user
+    if (documentResponse.data.compressionAlgorithm === 'GZIP') {
+      rawBuffer = await zlib.gunzipSync(contentResponse.data)
```

Comment:
It feels to me like gunzipSync and an await are at odds? Which one is it?

Reply:
You're right it can be removed, I had made a promise wrapper to handle the `zlib.gunzip` function which uses callbacks and used that with await here until I realized `gunzipSync` existed